src/server.rs

```diff
@@ -1,105 +1,290 @@
-use std::net::SocketAddr;
-use std::ops::DerefMut;
-use std::sync::{Arc, Mutex};
-
-use env_logger::Env;
-use tokio;
-use tokio::codec::Decoder;
-use tokio::io;
-use tokio::net::TcpListener;
-use tokio::prelude::*;
-use tokio::runtime;
-
-use crate::codec::ABCICodec;
-use crate::messages::abci::*;
-use crate::Application;
-
-/// Creates the TCP server and listens for connections from Tendermint
-pub fn serve<A>(app: A, addr: SocketAddr) -> io::Result<()>
+#[cfg(unix)]
+use std::path::PathBuf;
```

Comment:
as the whole thing is breaking, perhaps this doesn't need to be feature guarded and TCP vs IPC can have a prefix, just like in Tendermint configurations?